pkg/client/client.go

```diff
@@ -510,3 +545,21 @@ func (client *DataSetClient) onBufferDrop(buf *buffer.Buffer, status uint32, err
 		)...,
 	)
 }
+
+func (client *DataSetClient) constructUserAgent() {
```

Comment:
construct evoques that it will return some product (of some construction), while this methods configures an attribute based on other attributes
```suggestion
func (client *DataSetClient) configureUserAgent() {
```